tutorials/migrating-rails-app-from-heroku-to-google-appengine.md

```diff
@@ -6,207 +6,206 @@ tags: App Engine, Ruby, Rails, Heroku, Migration
-    Subsequent steps will assume that `cloud_sql_proxy` is in your `PATH`.
+1.  [Install Cloud SQL Proxy](https://cloud.google.com/sql/docs/postgres/sql-proxy#install).
+    Ensure that `cloud_sql_proxy` is in your environment's `PATH` variable.
 
-3.  The proxy will create local unix sockets connected to your database
-    instances. In your Rails app directory, create a directory to hold those
-    unix sockets.
+1.  The proxy creates local sockets connected to your database instances. In your Rails app
+    directory, create a directory to hold those sockets:
 
         cd /path/to/my/rails/app
         mkdir -p tmp/cloudsql
 
-4.  And start the proxy:
+1.  To start the proxy, run the following command:
 
         cloud_sql_proxy -dir=tmp/cloudsql
 
-    Note this will run the proxy in the foreground, so subsequent commands
-    would need to be run in a separate shell. If you prefer, feel free to
+    Note: This runs the proxy in the foreground, so subsequent commands
+    need to be run in a separate shell. If you prefer, feel free to
     background the process instead.
 
-5.  The Cloud SQL Proxy tool opens a unix socket in the directory
-    `/path/to/my/rails/app/tmp/cloudsql/${DB_CONNECTION_NAME}/`. You can then
+5.  The proxy opens a socket in the directory
+    `/path/to/my/rails/app/tmp/cloudsql/[CONNECTION-NAME]/`. You can then
     connect to the database simply by pointing at that socket. If you like, you
-    can test connecting with psql:
+    can test connecting with `psql:`
 
-        psql -h /path/to/my/rails/app/tmp/cloudsql/${DB_CONNECTION_NAME} \
+        psql -h /path/to/my/rails/app/tmp/cloudsql/[CONNECTION-NAME] \
           -U postgres
 
 ### Copy data from Heroku to Cloud SQL
 
 Now that you have a way to connect directly to your database instance, you can
-copy the data over from heroku.
+copy the data over from your Heroku database.
 
-1.  Dump the contents of the heroku database:
+1.  Export the contents of the Heroku database:
 
         heroku pg:backups:capture
         heroku pg:backups:download
 
-    Those commands should create a file `latest.dump` in the current directory.
+    Those commands create a file `latest.dump` in the current working directory.
 
-2.  Restore the data to your Cloud SQL database. Use the socket provided by
-    the Cloud SQL Proxy to connect to the database instance.
+1.  Restore the data to your Cloud SQL database. Use the socket provided by
+    the proxy to connect to the database instance:
 
         pg_restore --verbose --clean --no-acl --no-owner -d mydb -U postgres \
-          -h /path/to/my/rails/app/tmp/cloudsql/${DB_CONNECTION_NAME} \
+          -h /path/to/my/rails/app/tmp/cloudsql/[CONNECTION-NAME] \
           latest.dump
 
-There will probably be some warnings during the migration, but if there are no
-outright errors, your database should now be migrated!
+    Note: You might encounter permissible warnings during the migration.
```

Comment:
What's a "permissible warning"? I'd prefer a clearer explanation, because there are typically a lot of them, and it can be concerning to a user.

Reply:
In other words, warnings that can be ignored. How's this?:

"Note: You might encounter several warnings during the migration. Unless they result in errors, you can safely ignore these warnings."

Reply:
I didn't add this one myself, feel free to do so.

Reply:
Done